src/pages/Survey/index.jsx

```diff
@@ -1,21 +1,66 @@
 import style from './style.module.css'
-import OpenText from '../../components/OpenText'
+import axios from 'axios'
+import React from 'react'
+
+import Matrix from '../../components/Matrix'
 import MultipleChoice from '../../components/MultipleChoice'
+import OpenText from '../../components/OpenText'
+import { Button } from 'antd'
+
+const test = require('./test.json');
+
+const { REACT_APP_API_URL } = process.env
+
+function Survey({ id, ...props }) {
+  const [data, setData] = React.useState([''])
+  const [currQuestion, setCurrQuestion] = React.useState(0)
+  const [questionAnswers, setQuestionAnswers] = React.useState({})
+
+  const nextQuestion = () => {
+    setCurrQuestion((prev) =>
+      data.length - 1 > prev ? prev + 1 : prev
+    )
+  }
+
+  const prevQuestion = () => {
+    setCurrQuestion((prev) => (prev > 0 ? prev - 1 : prev))
+  }
 
-import data from '../../dataFromAPI/formattedQuestions.json'
+  const saveAnswer = (answer) => {
+    setQuestionAnswers({ ...questionAnswers, [currQuestion]: answer })
+    console.log(answer);
+  }
 
-function Survey(props) {
+  React.useEffect(() => {
+    axios.get(REACT_APP_API_URL + '/api/client/survey/' + id).then(setData)
+  }, [])
   return (
     <div className={style.survey}>
-      {data.questionnaire.map((question, idx) => {
-        if (question.type === 'multiple_choice') {
-          return <MultipleChoice data={question} key={idx} />
-        }
-        if (question.type === 'open_text') {
-          return <OpenText data={question} key={idx} />
-        }
-        return null
-      })}
+      {
+        data[currQuestion].type == 'matrix' ? (
+          <Matrix {...data[currQuestion]} setAnswers={saveAnswer} startingQuestionAnswers={questionAnswers[currQuestion] || {}} />
+        ) : data[currQuestion].type == 'multiple_choice' ? (
+          <MultipleChoice data={data[currQuestion]} setAnswer={saveAnswer} answers={questionAnswers[currQuestion]} />
+        ) : data[currQuestion].type == 'open_text' ? (
+          <OpenText data={data[currQuestion]} setAnswer={saveAnswer} answers={questionAnswers[currQuestion]} />
+        ) : null
+      }
```

Comment:
connecting ternary operators together is considered very bad practice, so instead just check for the type and use `&&`